packages/melange.js/Js.ml

```diff
@@ -757,7 +757,29 @@ end = struct
     if start_idx >= end_idx then ""
     else Stdlib.String.sub str start_idx (end_idx - start_idx)
 
-  let split ?sep ?limit _str = notImplemented "Js.String" "split"
+  let split ?sep ?limit str =
+    match%platform () with
+    | Client -> Js.String.split ~sep ~limit str
+    | Server -> (
+        let sep = Option.value sep ~default:str in
+        let regexp = Str.regexp_string sep in
+        let items =
+          match
+            ( Str.string_match regexp str 0,
+              Str.string_match regexp str (String.length str - String.length sep),
+              sep <> "" )
+          with
+          | true, true, true -> ("" :: Str.split regexp str) @ [ "" ]
+          | true, false, true -> "" :: Str.split regexp str
```

Comment:
Could you add comments to those cases? They look either edge-cases or rarities of JS

Reply:
Done: https://github.com/ml-in-barcelona/server-reason-react/pull/168/commits/b72485f7da48648848c435a87cdf79d1d185993f